Gordon360/ApiControllers/MyScheduleController.cs

```diff
@@ -0,0 +1,244 @@
+        private IScheduleControlService _scheduleControlService;
+
+
+        private IMyScheduleService _myScheduleService;
+
+        public MyScheduleController()
+        {
+            var _unitOfWork = new UnitOfWork();
+            _myScheduleService = new MyScheduleService(_unitOfWork);
+            _profileService = new ProfileService(_unitOfWork);
+            _accountService = new AccountService(_unitOfWork);
+            _roleCheckingService = new RoleCheckingService(_unitOfWork);
+            _scheduleControlService = new ScheduleControlService(_unitOfWork);
+        }
+
+        public MyScheduleController(IMyScheduleService myScheduleService)
+        {
+            _myScheduleService = myScheduleService;
+        }
+
+        /// <summary>
+        ///  Gets all custom events for a user
+        /// </summary>
+        /// <returns>A IEnumerable of custom events</returns>
+        [HttpGet]
+        [Route("")]
+        public IHttpActionResult Get()
+        {
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
+
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+
+            object result = _myScheduleService.GetAllForID(id);
+            if (result == null)
+            {
+                return NotFound();
+            }
+
+            JArray jresult = JArray.FromObject(result);
+
+            foreach (JObject elem in jresult)
+            {
+                elem.Property("GORDON_ID").Remove();
+            }
+
+            return Ok(jresult);
+        }
+
+
+        /// <summary>
+        ///  Gets specific custom event for a user
+        /// </summary>
+        /// <returns>The requested custom event</returns>
+        [HttpGet]
+        [Route("event/{event_id}")]
+        public IHttpActionResult GetByEventId(string event_Id)
+        {
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
+
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+
+            object result = _myScheduleService.GetForID(event_Id, id);
+            if (result == null)
+            {
+                return NotFound();
+            }
+
+            JObject jresult = JObject.FromObject(result);
+
+                jresult.Property("GORDON_ID").Remove();
+
+            return Ok(jresult);
+        }
+
+        /// <summary>
+        ///  Gets all myschedule objects for a user
+        /// </summary>
+        /// <returns>A IEnumerable of myschedule objects</returns>
+        [HttpGet]
+        [Route("{username}")]
+        public IHttpActionResult Get(string username)
+        {
+            //probably needs privacy stuff like ProfilesController and service
```

Comment:
I believe the privacy ended up being in the ScheduleService, but it would be good to test on postman to make sure this route can't be abused.